docs/src/main/asciidoc/application-configuration-guide.adoc

```diff
@@ -158,6 +158,144 @@ You can add the `greeting.suffix`, remove the other properties, change the value
 As usual, the application can be packaged using `./mvnw clean package` and executed using the `-runner.jar` file.
 You can also generate the native executable with `./mvnw clean package -Pnative`.
 
+== Using @ConfigProperties
+
+As an alternative to injecting multiple related configuration values in the way that was shown in the previous example,
+users can also use the `@io.quarkus.arc.runtime.config.ConfigProperties` annotation to group these properties together.
```

Comment:
I don't know about Arc but in the other extensions, everything that is API is at the root of the extension package and the `runtime` package is for the internal classes.

@mkouba ?

Reply:
@geoand we need to sort that out before merging as this annotation will be API.

Reply:
The problem is that the package `io.quarkus.arc` is used in the `io.quarkus.arc:arc` artifact and we would like to avoid a package clash to prevent problems with Java 9+. So we could either rename the package in `io.quarkus.arc:arc` or use a different package in the runtime module. E.g. `io.quarkus.arc.api`...

Reply:
Whatever you folks think is best. It's obviously no trouble for me to move the annotation